tests/harness/src/runner.rs

```diff
@@ -0,0 +1,136 @@
+use std::{
+    path::{Path, PathBuf},
+    process::Command,
+    time::Instant,
+};
+
+use anyhow::{Context, Result};
+use ceresdb_client_rs::client::Client;
+use tokio::{
+    fs::{remove_file, File, OpenOptions},
+    io::AsyncWriteExt,
+};
+use walkdir::WalkDir;
+
+use crate::case::TestCase;
+
+/// Input SQL query cases
+const TEST_CASE_EXTENSION: &'static str = "sql";
+/// Output generated by running queries
+const OUTPUT_FILE_EXTENSION: &'static str = "out";
+/// Expected result provides with queries
+const RESULT_FILE_EXTENSION: &'static str = "result";
+/// Tools to compare file
+const DIFF_BINARY: &'static str = "diff";
```

Comment:
There is a prettydiff, it has color support, which is useful to review differences.
- https://github.com/romankoblov/prettydiff

Reply:
I don't intend to keep the diff content. If the result and expected are the same it will automatically remove the output. Otherwise the output will be kept and reported. So one only needs to check whether there is any new ".out" file generated after a run. 

If we wants to inspect which lines are diff we can use whatever diff tools we want. Another reason that not reporting the diff content directly is to keep the ".out" and ".result" files have the same format.

Reply:
> So one only needs to check whether there is any new ".out" file generated after a run.

When multiple testcases failed, How one can check all bad diff? `cat` output file one by one is kinds of inconvenient.

Reply:
When multiple cases failed one may also want to fix them one by one. Mix up all the different together is unnecessary as the cases are organized by scenario. Failures in one file have a clear context, those across multiple files don't have.